app/src/main/java/io/stormbird/wallet/service/TransactionsNetworkClient.java

```diff
@@ -197,4 +200,42 @@ private String readTransactions(NetworkInfo networkInfo, String address, String
 			return result;
 		}).subscribeOn(Schedulers.io());
 	}
+
+	@Override
+	public Single<Integer> checkConstructorArgs(NetworkInfo networkInfo, String address)
+	{
+		return Single.fromCallable(() -> {
+			int result = 256;
+			try
+			{
+				String response = readTransactions(networkInfo, address, "0", true, 1, 5);
+
+				if (response != null)
+				{
+					JSONObject stateData = new JSONObject(response);
+					JSONArray orders = stateData.getJSONArray("result");
+					EtherscanTransaction[] myTxs = gson.fromJson(orders.toString(), EtherscanTransaction[].class);
+					for (EtherscanTransaction etx : myTxs)
+					{
+						Transaction tx = etx.createTransaction(null);
+						if (tx.isConstructor && tx.operations.length > 0)
+						{
+							TransactionContract ct = tx.operations[0].contract;
+							result = ct.decimals;
+							break;
+						}
+					}
+				}
+			}
+			catch (JSONException e)
+			{
```

Comment:
Both silently fail!

Reply:
correct. This happens, and we want a silent fail when we get a bad return from etherscan which can happen due to no fault of the app.